frontend/src/metabase/admin/settings/components/UploadSettings/UploadSettingsForm.tsx

```diff
@@ -284,12 +287,27 @@ export function UploadSettingsFormView({
   );
 }
 
-const H2PersistenceWarning = () => (
+const H2PersistenceWarning = ({ isHosted }: { isHosted: boolean }) => (
   <Stack my="md" maw={620}>
     <Alert icon="warning" variant="warning">
       <Text>
         {t`Warning: uploads to the Sample Database are for testing only and may disappear. If you want your data to stick around, you should upload to a PostgreSQL or MySQL database.`}
       </Text>
+      {isHosted && (
+        <Tooltip
+          label={
+            <>
+              <Text mb="md">{t`By enabling uploads to the Sample Database, you agree that you will not upload or otherwise transmit any individually identifiable information, including without limitation Personal Data (as defined by the General Data Protection Regulation) or Personally Identifiable Information (as defined by the California Consumer Privacy Act and California Privacy Rights Act).`}</Text>
+              <Text>{t`Additionally, you acknowledge and agree that the ability to upload to the Sample Database is provided “as is” and without warranty of any kind, and Metabase disclaims all warranties, express or implied, and all liability in connection with the uploads to the Sample Database or the data stored within it.`}</Text>
+            </>
+          }
+          position="bottom"
+          multiline
+          maw="30rem"
+        >
+          <Text span underline weight={700}>{t`Additional terms apply.`}</Text>
+        </Tooltip>
+      )}
```

Comment:
Not totally sure, but I think this should be an enterprise plugin? @luizarakaki ?

Reply:
If you take a look at the `getIsHosted` selector, we're reading this directly from the settings, not from the token features.

Reply:
ok, no need to make this an enterprise plugin! my bad!